### PR TITLE
fix: report flaky tests via build status to CI health

### PR DESCRIPTION
**Pull request by cmur2**
## Description

Report flaky tests from Unified CI & Zeebe CI [via build status](https://github.com/camunda/infra-global-github-actions/tree/main/submit-build-status) for CI health observability following the conventions for `build_status` and `user_reason`. This works towards the goal of #18210 and implicitly addresses some tasks from #18856  as well for the Zeebe CI.

The list of flaky tests is best-effort just for information until https://github.com/camunda/team-infrastructure/issues/608 gets implemented.

The data is ingested successfully (at the example of this PR and [this failed run](https://github.com/camunda/camunda/actions/runs/9990809809/job/27612279255)) and looks like this which can be later used for visualizations:

image

## Related issues

Related to #18210 and to #18856

### CODE REVIEW

**Comment by cmur2** [comment]
@lenaschoenburg @rodrigo-lourenco-lopes @houssain-barouni would one of you feel comfortable to review this PR? Due to absences my usual go-to people for these kinds of changes are OoO.

File: .github/workflows/zeebe-ci.yml [hunk 1]
```diff
@@ -157,10 +157,13 @@ jobs:
         uses: ./.github/actions/observe-build-status
         with:
           job_name: "integration-tests/${{ matrix.group }}"
-          build_status: ${{ contains(steps.*.conclusion, 'failure') && 'failure' || 'success' }}
+          build_status: ${{ job.status }}
```
Comment (lenaschoenburg) [inline .github/workflows/zeebe-ci.yml:160]: :thought_balloon: Haven't double checked if that works (i.e. that `job.status` is "failure" if any of the previous steps failed) but I'd trust you on this

**Review by lenaschoenburg — APPROVED** [review]
That's a really nice and easy improvement :heart_eyes:

File: .github/workflows/zeebe-ci.yml (hunk 1 above)
Comment (cmur2) [reply .github/workflows/zeebe-ci.yml:160]: Oh yes that is a copy&pasta from the Unified CI [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml#L89) where I introduced it to test what would happen (when I created it) and it has proven to work 🙂

**Pull request merged**